test/upload-download.test.js

```diff
@@ -91,6 +91,7 @@ describe('storage service', function() {
       .set('Content-Type', 'application/json')
       .expect('Content-Type', /json/)
       .expect(200, function(err, res) {
+        if (err) done(err);
```

Comment:
If you don't put the return, it doesn't short circuit at that line. You need it in this case to stop execution unless it's the last line in the func.